File: app/src/main/java/kisan/com/fabproject/Main2Activity.java
package kisan.com.fabproject;

import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.List;

public class Main2Activity extends AppCompatActivity {
    RecyclerView recyclerView;
    Main2ViewModel main2ViewModel ;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
        main2ViewModel = ViewModelProviders.of(this).get(Main2ViewModel.class);
        init();
        main2ViewModel.setIntent(getIntent());
    }

    private void init() {
        recyclerView = findViewById(R.id.recycler_view);

        setRecyclerView();
    }

    private void setRecyclerView() {
        main2ViewModel.recycleList.observe(this, new Observer<List<ForcastModel>>() {
            @Override
            public void onChanged(@Nullable List<ForcastModel> forcastModels) {
                ForcastAdapter adapter = new ForcastAdapter(forcastModels);
                LinearLayoutManager linearLayoutManager = new LinearLayoutManager(Main2Activity.this,LinearLayoutManager.VERTICAL,false);
                recyclerView.setLayoutManager(linearLayoutManager);
                recyclerView.setAdapter(adapter);
            }
        });
    }
}

File: app/src/main/java/kisan/com/fabproject/Main2Activity.kt
package kisan.com.fabproject

import android.app.FragmentTransaction
import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import kotlinx.android.synthetic.main.activity_main2.*

class Main2Activity : AppCompatActivity() {
    lateinit var main2ViewModel: Main2ViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
        main2ViewModel = ViewModelProviders.of(this).get(Main2ViewModel::class.java)
        init()
        main2ViewModel.setIntent(intent)
    }

    private fun init() {

        setView()
    }


// seting observer for showing data to user
    private fun setView() {
        main2ViewModel.currentDataModel.observe(this, Observer { currentDataModel ->
            currentDataModel?.let {
                temp.text = currentDataModel.temp.toFloat().toInt().toString() + "°"
                name.text = currentDataModel.city
                Handler(Looper.getMainLooper()).postDelayed(Runnable {
                    val ft = supportFragmentManager.beginTransaction();
                    ft.setCustomAnimations(R.anim.translate,0)
                    ft.add(R.id.frame,ForcastFrgament.newInstance(it.list)).commit()
                },500)
            }

        })
    }
}
